planner/aamwriter/src/main/java/eu/seaclouds/platform/planner/aamwriter/modeldesigner/DRequirements.java

```diff
@@ -42,13 +44,22 @@ public DRequirements(JSONObject jnode, DGraph graph) {
         this.availability = parseDouble(jnode, Attributes.AVAILABILITY);
         this.cost = parseDouble(jnode, Attributes.COST);
         this.workload = parseDouble(jnode, Attributes.WORKLOAD);
+        this.frontendModuleName = parseString(jnode, Attributes.FRONTEND);
     }
     
     private double parseDouble(JSONObject jnode, String key) {
-        double result = Double.parseDouble(jnode.get(key).toString());
+        Object value = jnode.get(key);
+        double result = (value == null)? 0: Double.parseDouble(value.toString());
         
         return result;
     }
+    
+    private String parseString(JSONObject jnode, String key) {
+        Object value = jnode.get(key);
+        
+        String result = (value == null)? "" : value.toString();
+        return result;

```

Comment:
No. You can define the method in one line, or not even using a method at all. But if you need to debug the method to see what value is taking each step, you will need to modify the code. I also prefer to have a single return point, with no conditions in it.